File: message_stats.py
#!/usr/bin/env python3

# INSTAGRAM JSON VIEWER - Message Statistics
# 2020 (c) Micha Johannes Birklbauer
# https://github.com/t0xic-m/
# micha.birklbauer@gmail.com

import pandas as pd
import json

# only 1-1 conversations, no groupchats
def get_stats(username):
    stats = {}

    with open("messages.json", "r", encoding="utf-8") as f:
        data = json.load(f)
        f.close()

    for conversation in data:
        participants = conversation["participants"]
        if len(participants) != 2:
            continue
        partner_1 = participants[0]
        partner_2 = participants[1]
        if partner_1 == username:
            if partner_2 in stats:
                stats[partner_2] = stats[partner_2] + len(conversation["conversation"])
            else:
                stats[partner_2] = len(conversation["conversation"])
        else:
            if partner_1 in stats:
                stats[partner_1] = stats[partner_1] + len(conversation["conversation"])
            else:
                stats[partner_1] = len(conversation["conversation"])

    df = pd.DataFrame({"conversation":list(stats.keys()), "messages":list(stats.values())})
    df.to_csv("message_stats.csv")

    return df

if __name__ == '__main__':
    username = input("Enter your Instagram username:\n")
    result = get_stats(username)

File: message_stats.R
#!/usr/bin/env Rscript

# INSTAGRAM JSON VIEWER - Message Statistics
# 2020 (c) Micha Johannes Birklbauer
# https://github.com/t0xic-m/
# micha.birklbauer@gmail.com

library(ggplot2)

df = read.csv2("message_stats.csv", sep = ",", dec = ".")

ggplot(df, aes(x=reorder(conversation, -messages), y=messages)) + 
  geom_bar(stat="identity", color="black", fill="dodgerblue3", width = 0.7) + 
  ggtitle("Number of Messages per Contact") + 
  xlab("Conversation Partner (Instagram Username)") + 
  ylab("Number of Messages") + 
  geom_text(aes(label=messages), vjust=-0.3, size=3.0) + theme_minimal() + 
  theme(axis.text.x = element_text(angle = 90, hjust = 1))

ggsave("message_stats.png")
ggsave("message_stats.jpeg")